src/main.cpp

```diff
@@ -885,21 +885,25 @@ class GraspProcessorModule : public RFModule
                     candidate_pose->pose_rotation = candidate_pose->pose_rotation * yarp::math::axis2dcm(y_rotation_transform).submatrix(0, 2, 0, 2);
                     candidate_pose->pose_translation = superq_center + (s * candidate_pose->pose_ax_size(2) / norm(gz)) * gz - (0.01 / norm(gx)) * gx;
 
+                    Vector table_height_corr = superq_center;
+                    table_height_corr[2] = table_height_z;
+                    fixReachingOffset(table_height_corr,table_height_corr);
+
                     if (isCandidateGraspFeasible(candidate_pose))
                     {
                         //  if the grasp is close to the table surface, we need to adjust the pose to avoid collision
-                        bool side_low = is_side_grasp && ((candidate_pose->pose_translation(2) - palm_width) < table_height_z);
-                        bool top_low = !is_side_grasp && ((candidate_pose->pose_translation(2) - finger_length) < table_height_z);
+                        bool side_low = is_side_grasp && ((candidate_pose->pose_translation(2) - 0.4 * palm_width) < table_height_corr[2]);
+                        bool top_low = !is_side_grasp && ((candidate_pose->pose_translation(2) - 0.9 * finger_length) < table_height_corr[2]);
 
                         if (side_low)
                         {
                             //  lift up the grasp closer to the upper end of the superquadric
-                            candidate_pose->pose_translation(2) = table_height_z + 0.6 * palm_width;
+                            candidate_pose->pose_translation(2) = table_height_corr[2] + 0.6 * palm_width;
```

Comment:
@fbottarel the couple of coefficients `[0.4 0.6]` for the palm width has been chosen to be conservative: if we go under the table for a length shorter than half palm (i.e. `0.4`), then go up of a bit more of half palm (i.e. `0.6`).